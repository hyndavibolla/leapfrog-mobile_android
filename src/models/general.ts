import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
