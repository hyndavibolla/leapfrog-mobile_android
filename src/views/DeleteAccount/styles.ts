import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '_constants';

export const styles = {
  container: css`
    background-color: ${COLOR.LIGHT_GRAY};
    height: 100%;
    padding: 0 40px;
    flex: 1;
    justify-content: center;
    align-items: center;
  `,
  deleteIcon: css`
    width: 48px;
    height: 48px;
  `,
  title: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.MEDIUM};
    text-align: center;
    margin-top: 16px;
  `,
  textContent: css`
    margin-top: 8px;
    font-family: ${FONT_FAMILY.MEDIUM};
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
    text-align: center;
  `,
  email: css`
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
    color: ${COLOR.BLACK};
    line-height: ${FONT_SIZE.REGULAR};
    margin-top: 32px;
    text-align: center;
  `,
  link: css`
    margin-top: 16px;
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.PETITE};
    line-height: 22px;
    color: ${COLOR.PRIMARY_BLUE};
    text-align: center;
    margin-bottom: 16px;
  `
};
