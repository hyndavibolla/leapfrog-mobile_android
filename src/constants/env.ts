// @ts-nocheck
import {
  ENVIRONMENT,
  BRANCH_NAME,
  API_URL,
  API_RETRY_REQUEST_ATTEMPTS,
  API_RETRY_REQUEST_ATTEMPT_DELAY_MS,
  API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX,
  SPLASH_MIN_ANIMATION_TIME_MS,
  SCHEME,
  WEBVIEWS_MAX_LOAD_ATTEMPTS,
  WEBVIEWS_ATTEMPT_DELAY_MS,
  SYW_URL,
  SYW_APP_PROFILE,
  SYW_SUPPORT_EMAIL,
  FUSION_APPLY_NOW_WEBVIEW_URI,
  FUSION_APPLICATION_FINISHED_URI_REGEX,
  API_TOKEN_EXPIRATION_PADDING_MS,
  IGNORED_FEATURE_LIST,
  LOCALE,
  TOAST_VISIBLE_MS,
  ACTIVITY_FETCH_HISTORY_MONTH_DELTA,
  LOCAL_OFFERS_DEFAULT_ACTIVATION_DAYS,
  LOCAL_OFFERS_ENROLLMENT_FINISH_URI_REGEX,
  MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA,
  MERCHANT,
  MIN_ACTIVITY_FETCH_HISTORY_RESULTS,
  MISSION_KEEP_EARNING_LIMIT,
  MISSION_FETCH_LIMIT,
  CITI_LANDING_URL,
  CITI_PHONE_NUMBER,
  MODAL_PADDING_MS,
  API_CACHE_GAME_STATE_MS,
  API_CACHE_STREAK_LIST_MS,
  EARN_POINTS_ANIMATION_MS,
  SLIDE_URL,
  SLIDE_CLIENT_ID,
  SLIDE_CLIENT_SECRET,
  SLIDE_CHECKOUT_BASE_URL,
  SLIDE_CHECKOUT_PATH,
  SLIDE_CHECKOUT_CLIENT_HEADER,
  SLIDE_APPLICATION_ID,
  SLIDE_APPLICATION_FINISHED_URI_PREFIXES,
  GIFT_CARD_MAX_OPTIONS_COUNT,
  BUTTON_APP_ID_ANDROID,
  BUTTON_APP_ID_IOS,
  TEALIUM_ACCOUNT,
  TEALIUM_PROFILE,
  SPINNER_MESSAGE_CHANGE_MS,
  CONNECTION_LOSS_PAUSE_TIMEOUT,
  ONE_TRUST_ANDROID_ID,
  ONE_TRUST_ANDROID_URL,
  ONE_TRUST_IOS_ID,
  ONE_TRUST_IOS_URL,
  MISSION_FEATURED_LIMIT,
  MISSION_SPECIAL_LIMIT,
  MISSION_JUST_FOR_YOU_LIMIT,
  MISSION_TRENDING_LIMIT,
  MISSION_STREAK_LIMIT,
  IMAGE_LOADING_FLICKERING_DEBOUNCE_MS,
  MISSION_SEARCH_HISTORY_LENGTH,
  MISSION_SEARCH_PREDICTIVE_LENGTH,
  PURCHASE_MODAL_TIMEOUT_MS,
  APPSFLYER_DEV_KEY,
  APPSFLYER_APP_ID_ANDROID,
  APPSFLYER_APP_ID_IOS,
  REMOTE_CONFIG_KEY_APP_VERSION,
  REMOTE_CONFIG_CACHE_MS,
  REMOTE_CONFIG_EARN_SECTIONS,
  APP_STORE_URL,
  PLAY_STORE_URL,
  MISSION_TRACKING_TIME_THRESHOLD,
  MISSION_TRACKING_VISIBILITY_THRESHOLD,
  PN_DAYS_TO_REQUEST_PERMISSION,
  LOGIN_REFRESH_INTERVAL_MS,
  POINTS_EXPIRE_BANNER_THRESHOLD_LIST,
  FORTER_SITE_ID,
  STORAGE_CLEAN_WAIT,
  AUTH0_DOMAIN_ID,
  AUTH0_CLIENT_ID,
  AUTH0_AUDIENCE,
  AUTH0_SCOPE,
  TRACK_EVENT_DELAY_MS,
  LOCAL_OFFERS_SEARCH_DEBOUNCE_MS,
  TELLURIDE_API_CONTENT_BASE_URL,
  FORTER_FRAUD_PREVENTION_URL,
  FORTER_DENIED_ACCESS_URL_REGEX,
  FORTER_ALLOWED_ACCESS_URL_REGEX,
  TUTORIAL_HIDE_DELAY_MS,
  TUTORIAL_CONGRATS_DELAY_MS,
  TUTORIAL_END_DELAY_MS,
  TUTORIAL_BANNER_CONGRATS_DELAY_MS,
  TUTORIAL_BANNER_CHECK_DELAY_MS,
  TUTORIAL_BANNER_CHECK_DELAY_FROM_SKIP_MS,
  AB_IOS_NEW_NON_ECM_BANNER_DEV
} from '@env';
const appJson = require('../../app.json'); // tslint:disable-line
const packageJson = require('../../package.json'); // tslint:disable-line
import { FeatureFlag } from '../models/general';
export const ENV = {
  APP_NAME: appJson.displayName,
  APP_VERSION: packageJson.version,
  ENVIRONMENT,
  IS_TEST: ENVIRONMENT === 'test',
  IS_DEV: ENVIRONMENT === 'dev',
  IS_QA: ENVIRONMENT === 'qa',
  IS_UAT: ENVIRONMENT === 'uat',
  IS_PROD: ['uat', 'prod'].includes(ENVIRONMENT),
  BRANCH_NAME,
  SCHEME: SCHEME,
  API: {
    URL: API_URL,
    API_TOKEN_EXPIRATION_PADDING_MS: Number(API_TOKEN_EXPIRATION_PADDING_MS),
    API_RETRY_REQUEST_ATTEMPTS: Number(API_RETRY_REQUEST_ATTEMPTS),
    API_RETRY_REQUEST_ATTEMPT_DELAY_MS: Number(API_RETRY_REQUEST_ATTEMPT_DELAY_MS),
    API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX: new RegExp(API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX),
    CACHE: {
      GAME_STATE_MS: Number(API_CACHE_GAME_STATE_MS),
      STREAK_LIST_MS: Number(API_CACHE_STREAK_LIST_MS)
    },
    SLIDE: {
      URL: SLIDE_URL,
      CLIENT_ID: SLIDE_CLIENT_ID,
      CLIENT_SECRET: SLIDE_CLIENT_SECRET,
      CHECKOUT_BASE_URL: SLIDE_CHECKOUT_BASE_URL,
      CHECKOUT_PATH: SLIDE_CHECKOUT_PATH,
      CHECKOUT_CLIENT_HEADER: SLIDE_CHECKOUT_CLIENT_HEADER,
      APPLICATION_ID: SLIDE_APPLICATION_ID,
      FINISHED_URI_PREFIXES: String(SLIDE_APPLICATION_FINISHED_URI_PREFIXES)
    }
  },
  TEALIUM: {
    ACCOUNT: TEALIUM_ACCOUNT,
    PROFILE: TEALIUM_PROFILE
  },
  SPLASH_MIN_ANIMATION_TIME_MS: Number(SPLASH_MIN_ANIMATION_TIME_MS),
  /** @todo move these to a constants file, since these are not configurable */
  STORAGE_KEY: {
    DEVICE_ID: 'device-id-v2',
    TOKEN_SET: 'token-set',
    GAME_LEVEL_UI: 'game-level-user-interaction',
    GAME_CURRENT: 'state-game-current',
    CCPA_SETTING: 'ccpa-setting',
    COOKIE_REGISTRY: 'cookie-registry',
    ONBOARDING: 'onboarding',
    MISSION_SEARCH_HISTORY: 'mission-search-history',
    VIEWED_TOOLTIP_LIST: 'viewed-tooltip-list',
    DISMISSED_TOOLTIP_LIST: 'dismissed-tooltip-list',
    API_OVERRIDE_SETTINGS: 'api-override-settings',
    LOG_SETTINGS: 'log-settings',
    LAST_SUGGESTED_BUILD_VERSION: 'last-suggested-build-version',
    MISSION_AFTER_PURCHASE_MODAL: 'mission-after-purchase-modal',
    LAST_TIME_PN_PERMISSION_ASKED: 'last_time_pn_permission_asked',
    HAS_VISITED_MAIN_SECTION: 'has_visited_main_section',
    EXPIRE_POINTS_BANNER: 'expire-points-banner',
    AUTH0_TOKEN_SET: 'auth0-token-set',
    NEW_ON_MAX_LAST_ONBOARD: 'new-on-max-last-onboard',
    SHOW_OFFERS_ON_EARN_MAIN: 'show-offers-on-earn-main',
    LOCAL_OFFERS_SEARCH_HISTORY: 'local-offers-search-history',
    OFFERS_SEARCH_HISTORY: 'offers-search-history',
    SID: 'sid-tracking-value',
    FORCED_TO_LOGOUT: 'forced-to-logout',
    GIFT_CARDS_SEARCH_HISTORY: 'gc-search-history',
    TUTORIAL_SKIP_STATUS: 'tutorial-skip-status',
    CURRENT_LOCATION: 'current-location',
    RECENTLY_VIEWED_MISSIONS: 'recently-viewed-missions'
  },
  WEBVIEWS: {
    MAX_LOAD_ATTEMPTS: Number(WEBVIEWS_MAX_LOAD_ATTEMPTS),
    ATTEMPT_DELAY_MS: Number(WEBVIEWS_ATTEMPT_DELAY_MS)
  },
  FUSION: {
    FINISHED_URI_REGEX: String(FUSION_APPLICATION_FINISHED_URI_REGEX),
    APPLY_NOW_URI: String(FUSION_APPLY_NOW_WEBVIEW_URI)
  },
  LOCALE,
  IGNORED_FEATURE_LIST: String(IGNORED_FEATURE_LIST)
    .split(',')
    .map<FeatureFlag>(s => s.trim()),
  TOAST_VISIBLE_MS: Number(TOAST_VISIBLE_MS),
  ACTIVITY_FETCH_HISTORY_MONTH_DELTA: Number(ACTIVITY_FETCH_HISTORY_MONTH_DELTA),
  LOCAL_OFFERS: {
    DEFAULT_ACTIVATION_DAYS: Number(LOCAL_OFFERS_DEFAULT_ACTIVATION_DAYS),
    ENROLLMENT_FINISH_URI_REGEX: LOCAL_OFFERS_ENROLLMENT_FINISH_URI_REGEX,
    SEARCH_DEBOUNCE_MS: Number(LOCAL_OFFERS_SEARCH_DEBOUNCE_MS)
  },
  MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA: Number(MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA),
  MERCHANT: MERCHANT,
  MIN_ACTIVITY_FETCH_HISTORY_RESULTS: Number(MIN_ACTIVITY_FETCH_HISTORY_RESULTS),
  MISSION_LIMIT: {
    FULL: Number(MISSION_FETCH_LIMIT),
    KEEP_EARNING: Number(MISSION_KEEP_EARNING_LIMIT),
    FEATURED: Number(MISSION_FEATURED_LIMIT),
    SPECIAL: Number(MISSION_SPECIAL_LIMIT),
    JUST_FOR_YOU: Number(MISSION_JUST_FOR_YOU_LIMIT),
    TRENDING: Number(MISSION_TRENDING_LIMIT),
    STREAK: Number(MISSION_STREAK_LIMIT)
  },
  CITI: {
    LANDING_URL: CITI_LANDING_URL,
    PHONE_NUMBER: CITI_PHONE_NUMBER
  },
  MODAL_PADDING_MS: Number(MODAL_PADDING_MS),
  ANIMATION: {
    EARN_POINTS: Number(EARN_POINTS_ANIMATION_MS)
  },
  GIFT_CARD_MAX_OPTIONS_COUNT: Number(GIFT_CARD_MAX_OPTIONS_COUNT),
  BUTTON: {
    APP_ID_ANDROID: BUTTON_APP_ID_ANDROID,
    APP_ID_IOS: BUTTON_APP_ID_IOS
  },
  CUSTOM_BRAND_CATEGORY_KEY: 'POPULAR_BRANDS',
  BLACKLIST_BRAND_CATEGORY_KEY: 'BLACKLISTED',
  SPINNER_MESSAGE_CHANGE_MS: Number(SPINNER_MESSAGE_CHANGE_MS),
  SYW_URL: SYW_URL,
  SYW_APP_PROFILE: SYW_APP_PROFILE,
  SYW_SUPPORT_EMAIL: SYW_SUPPORT_EMAIL,
  CONNECTION_LOSS_PAUSE_TIMEOUT: Number(CONNECTION_LOSS_PAUSE_TIMEOUT),
  ONE_TRUST: {
    ANDROID: {
      ID: ONE_TRUST_ANDROID_ID,
      URL: ONE_TRUST_ANDROID_URL
    },
    IOS: {
      ID: ONE_TRUST_IOS_ID,
      URL: ONE_TRUST_IOS_URL
    }
  },
  IMAGE_LOADING_FLICKERING_DEBOUNCE_MS: Number(IMAGE_LOADING_FLICKERING_DEBOUNCE_MS),
  MISSION_SEARCH_HISTORY_LENGTH: Number(MISSION_SEARCH_HISTORY_LENGTH),
  MISSION_SEARCH_PREDICTIVE_LENGTH: Number(MISSION_SEARCH_PREDICTIVE_LENGTH),
  MEMBER_NOT_FOUND_ERROR_CODE: 3053,
  MISSION_MAX_SELECTED_CATEGORIES: 1,
  PURCHASE_MODAL_TIMEOUT_MS: Number(PURCHASE_MODAL_TIMEOUT_MS),
  APPSFLYER: {
    DEV_KEY: APPSFLYER_DEV_KEY,
    APP_ID_ANDROID: APPSFLYER_APP_ID_ANDROID,
    APP_ID_IOS: APPSFLYER_APP_ID_IOS
  },
  REMOTE_CONFIG: {
    DEFAULT_CONFIG: { [REMOTE_CONFIG_KEY_APP_VERSION]: JSON.stringify({ suggested: 0, required: 0 }) },
    CACHE_TIMEOUT: Number(REMOTE_CONFIG_CACHE_MS),
    KEY: {
      APP_VERSION: REMOTE_CONFIG_KEY_APP_VERSION,
      EARN_SECTIONS: REMOTE_CONFIG_EARN_SECTIONS
    }
  },
  STORE_URL: {
    IOS: APP_STORE_URL,
    ANDROID: PLAY_STORE_URL
  },
  MISSION_TRACKING: {
    TIME_THRESHOLD: Number(MISSION_TRACKING_TIME_THRESHOLD),
    VISIBILITY_THRESHOLD: Number(MISSION_TRACKING_VISIBILITY_THRESHOLD)
  },
  PN_DAYS_TO_REQUEST_PERMISSION: Number(PN_DAYS_TO_REQUEST_PERMISSION),
  LOGIN_REFRESH_INTERVAL_MS: Number(LOGIN_REFRESH_INTERVAL_MS),
  POINTS_EXPIRE_BANNER_THRESHOLD_LIST: POINTS_EXPIRE_BANNER_THRESHOLD_LIST.split(',').map(Number) as number[],
  ROUTE_HISTORY_MAX_LENGTH: 50,
  FORTER_SITE_ID: FORTER_SITE_ID,
  STORAGE_CLEAN_WAIT: Number(STORAGE_CLEAN_WAIT),
  AUTH0: {
    DOMAIN_ID: AUTH0_DOMAIN_ID,
    CLIENT_ID: AUTH0_CLIENT_ID,
    AUDIENCE: AUTH0_AUDIENCE,
    SCOPE: AUTH0_SCOPE
  },
  SAILTHRU_MESSAGES_CATEGORIES: {
    NEW_ON_MAX: 'NEW_ON_MAX'
  },
  TRACK_EVENT_DELAY_MS: Number(TRACK_EVENT_DELAY_MS),
  TELLURIDE_API: {
    CONTENT_BASE_URL: TELLURIDE_API_CONTENT_BASE_URL
  },
  INACTIVE_DURATION_LIMIT_TIME: 5,
  FORTER_API: {
    FRAUD_PREVENTION_URL: FORTER_FRAUD_PREVENTION_URL,
    DENIED_ACCESS_URL_REGEX: FORTER_DENIED_ACCESS_URL_REGEX,
    ALLOWED_ACCESS_URL_REGEX: FORTER_ALLOWED_ACCESS_URL_REGEX
  },
  TUTORIAL: {
    HIDE_DELAY_MS: TUTORIAL_HIDE_DELAY_MS,
    CONGRATS_DELAY_MS: TUTORIAL_CONGRATS_DELAY_MS,
    END_DELAY_MS: TUTORIAL_END_DELAY_MS,
    BANNER: {
      CONGRATS_DELAY_MS: TUTORIAL_BANNER_CONGRATS_DELAY_MS,
      CHECK_DELAY_MS: TUTORIAL_BANNER_CHECK_DELAY_MS,
      CHECK_DELAY_FROM_SKIP_MS: TUTORIAL_BANNER_CHECK_DELAY_FROM_SKIP_MS
    }
  },
  EXPERIMENTS: {
    AB_IOS_NEW_NON_ECM_BANNER_DEV: AB_IOS_NEW_NON_ECM_BANNER_DEV
  }
};
