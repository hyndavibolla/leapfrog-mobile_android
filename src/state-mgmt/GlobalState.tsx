import React, { createContext, memo, PropsWithChildren, useReducer, Dispatch, useEffect } from 'react';

import { IGlobalState, Deps, IReducer, IGlobalContext } from '_models/general';
import { reducer as coreReducer } from './core/reducer';
import { reducer as userReducer } from './user/reducer';
import { reducer as gameReducer } from './game/reducer';
import { reducer as missionReducer } from './mission/reducer';
import { reducer as rewardReducer } from './reward/reducer';
import { reducer as surveyReducer } from './survey/reducer';
import { reducer as messagesReducer } from './messages/reducer';
import { reducer as cardLinkReducer } from './cardLink/reducer';
import { reducer as giftCardReducer } from './giftCard/reducer';
import { initialState as coreInitialState } from './core/state';
import { initialState as userInitialState } from './user/state';
import { initialState as gameInitialState } from './game/state';
import { initialState as missionInitialState } from './mission/state';
import { initialState as rewardInitialState } from './reward/state';
import { initialState as surveyInitialState } from './survey/state';
import { initialState as messagesInitialState } from './messages/state';
import { initialState as cardLinkInitialState } from './cardLink/state';
import { initialState as giftCardInitialState } from './giftCard/state';
import { combineReducers } from '_utils/combineReducers';
import { GeneralModel } from '_models';
import { deepClone } from '_utils/deepClone';

export const getInitialState = () =>
  deepClone<IGlobalState>({
    core: coreInitialState,
    user: userInitialState,
    game: gameInitialState,
    mission: missionInitialState,
    reward: rewardInitialState,
    survey: surveyInitialState,
    messages: messagesInitialState,
    cardLink: cardLinkInitialState,
    giftCard: giftCardInitialState
  });

export const combinedReducer = combineReducers<GeneralModel.IGlobalState>({
  core: coreReducer,
  user: userReducer,
  game: gameReducer,
  mission: missionReducer,
  reward: rewardReducer,
  survey: surveyReducer,
  messages: messagesReducer,
  cardLink: cardLinkReducer,
  giftCard: giftCardReducer
});

export const GlobalContext = createContext<{ state: IGlobalState; dispatch: Dispatch<any>; deps: Deps }>({
  state: getInitialState(),
  dispatch: /* istanbul ignore next */ () => null,
  deps: null
} as IGlobalContext);

export const GlobalProvider = memo(
  ({
    children,
    deps,
    initState,
    combinedReducers
  }: PropsWithChildren<{
    deps: Deps;
    initState: IGlobalState;
    combinedReducers: IReducer<IGlobalState>;
  }>) => {
    const [state, dispatch] = useReducer(combinedReducers, initState);
    useEffect(() => deps.stateSnapshot.set(state), [state]); // eslint-disable-line react-hooks/exhaustive-deps
    return <GlobalContext.Provider value={{ state, dispatch, deps } as IGlobalContext}>{children}</GlobalContext.Provider>;
  }
);
