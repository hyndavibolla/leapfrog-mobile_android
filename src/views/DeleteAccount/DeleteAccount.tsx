import React, { useCallback, useContext } from 'react';
import { View, Image, TouchableOpacity } from 'react-native';

import { Text } from '_components/Text';
import { styles } from './styles';
import { ENV, FONT_FAMILY, ForterActionType, PageType, TealiumEventType, UxObject } from '_constants';

import { GlobalContext } from '_state_mgmt/GlobalState';
import { useTestingHelper } from '_utils/useTestingHelper';
import { useEventTracker } from '_state_mgmt/core/hooks';

const subject = 'Account Deletion Request';

const DeleteAccount = () => {
  const {
    state: {
      user: {
        currentUser: {
          personal: { sywMemberNumber }
        }
      }
    },
    deps: {
      nativeHelperService: {
        linking: { openURL }
      }
    }
  } = useContext(GlobalContext);

  const { trackUserEvent } = useEventTracker();
  const { getTestIdProps } = useTestingHelper('delete-account');

  const onEditProfileOnSYW = useCallback(() => {
    trackUserEvent(
      TealiumEventType.PROFILE,
      {
        event_type: PageType.DELETE_SYW_ACCOUNT,
        uxObject: UxObject.BUTTON,
        event_detail: 'delete_account_email',
        exit_link: ENV.SYW_URL
      },
      ForterActionType.TAP
    );
    openURL(`mailto:${ENV.SYW_SUPPORT_EMAIL}?subject=${subject}&body=${sywMemberNumber} has requested that their account be deleted.`);
  }, [trackUserEvent, sywMemberNumber, openURL]);

  return (
    <View style={styles.container}>
      <Image style={styles.deleteIcon as any} source={require('_assets/shared/deleteAccountIcon.png')} />
      <Text font={FONT_FAMILY.HEAVY} style={styles.title}>
        We're sad you're leaving!
      </Text>
      <Text style={styles.textContent}>
        If you delete the account, all your information within Shop Your Way MAX will be removed, including points earned. This action can only be done by
        contacting our customer service.
      </Text>
      <Text font={FONT_FAMILY.HEAVY} style={styles.email}>
        E-mail:
      </Text>
      <TouchableOpacity onPress={onEditProfileOnSYW} {...getTestIdProps('support-email')}>
        <Text style={styles.link}>{ENV.SYW_SUPPORT_EMAIL}</Text>
      </TouchableOpacity>
    </View>
  );
};

export default DeleteAccount;
