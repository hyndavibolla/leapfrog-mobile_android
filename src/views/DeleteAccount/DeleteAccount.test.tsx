import React from 'react';
import DeleteAccount from './DeleteAccount';
import { fireEvent } from '@testing-library/react-native';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { Deps } from '_models/general';
import { ENV } from '_constants/env';
import { getMockDeps } from '_test_utils/getMockDeps';
import { getInitialState } from '_state_mgmt/GlobalState';

describe('DeleteAccount', () => {
  let initialState = getInitialState();
  let deps: Deps;

  beforeEach(() => {
    initialState.user.currentUser.personal.sywMemberNumber = '1234';
    deps = getMockDeps();
  });
  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<DeleteAccount />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should open contact support link', () => {
    const { getByTestId } = renderWithGlobalContext(<DeleteAccount />, deps, initialState);
    fireEvent.press(getByTestId('delete-account-support-email'));
    expect(deps.nativeHelperService.linking.openURL).toBeCalledWith(
      `mailto:${ENV.SYW_SUPPORT_EMAIL}?subject=Account Deletion Request&body=1234 has requested that their account be deleted.`
    );
  });
});
