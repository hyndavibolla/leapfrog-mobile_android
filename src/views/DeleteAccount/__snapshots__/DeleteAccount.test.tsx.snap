// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccount should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingBottom": 0,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 0,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/shared/deleteAccountIcon.png",
      }
    }
    style={
      Object {
        "height": 48,
        "width": 48,
      }
    }
  />
  <Text
    font="SFProDisplay-Heavy"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 16,
          "lineHeight": 22,
        },
        Object {
          "color": "#262C2F",
          "fontSize": 22,
          "marginTop": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    We're sad you're leaving!
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 16,
          "lineHeight": 22,
        },
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 16,
          "lineHeight": 24,
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
  >
    If you delete the account, all your information within Shop Your Way MAX will be removed, including points earned. This action can only be done by contacting our customer service.
  </Text>
  <Text
    font="SFProDisplay-Heavy"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 16,
          "lineHeight": 22,
        },
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 20,
          "marginTop": 32,
          "textAlign": "center",
        },
      ]
    }
  >
    E-mail:
  </Text>
  <View
    accessibilityLabel="delete-account-support-email"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="delete-account-support-email"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#0066CC",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 14,
            "lineHeight": 22,
            "marginBottom": 16,
            "marginTop": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      sywsolutions@syw.com
    </Text>
  </View>
</View>
`;
